File: AppBackEnd/AppBackEnd/Domain/Models/Cuestionario.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AppBackEnd.Domain.Models
{
    public class Cuestionario
    {
        public int Id { get; set; }

        [Required]
        [Column(TypeName = "varchar(100)")]
        public string Nombre { get; set; }

        [Required]
        [Column(TypeName = "varchar(150)")]
        public string Descripcion { get; set; }

        public DateTime FechaCreacion { get; set; }
        public int Activo { get; set; }

        public int UsuarioId { get; set; }
        public Usuario Usuario { get; set; }

        public List<Pregunta> listPreguntas { get; set; }
    }
}

File: appFrontEnd/src/app/models/cuestionario.ts
import { Pregunta } from './pregunta';
import { Usuario } from './usuario';

export class Cuestionario {
    id?: number;
    nombre: string;
    descripcion: string;
    fechaCreacion?: Date;
    listPreguntas?: Pregunta[];
    usuario ?: Usuario;

    constructor(nombre: string, descripcion: string, fechaCreacion: Date, listPreguntas: Pregunta[]){
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.fechaCreacion = fechaCreacion;
        this.listPreguntas = listPreguntas;
    }
}
